### DISCUSSION TITLE
Inconsistent history

### DISCUSSION THREAD

**Discussion by bhigy** [community/community#23673] [New to GitHub]
<p>I noticed some strange behavior on one of my repositories where the history is not consistent when I go back a couple of commits. To illustrate that, imagine the following recent history:</p>
<pre><code class="lang-auto">commit6
commit5
commit4
commit3
commit2
commit1
</code></pre>
<p>When I check the state of the history after commit 4 (from github website), I get:</p>
<pre><code class="lang-auto">commit4
commit1
</code></pre>
<p>Has anyone ever seen anything like that?</p>
<p>Best,<br>
Bertrand</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Are there any merges involved in the history? After a merge <code>git log</code> will sort commits from the merged branches by time, so it might look like the histories of the branches were mixed. But the branching history is still there, if you look at the history of an individual commit you’ll see only it’s ancestors.</p>
